clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerModule.h

```diff
@@ -91,6 +91,8 @@ class LowerModule {
   LogicalResult rewriteFunctionCall(CallOp callOp, FuncOp funcOp);
 };
 
+LowerModule createLowerModule(ModuleOp module, PatternRewriter &rewriter);
```

Comment:
I added it to this internal header. If we want to use it in, let's say, `LowerToLLVM.cpp`, we still have to write:

```cpp
#include "../../Dialect/Transforms/TargetLowering/LowerModule.h"
```

But I'm fine with leaving the interim state somehow dirty. It also encourage the eager refactor.

Reply:
That's the problem with technical debt, we have a FIXME on the other dirty inclusion and now it seems encouraging to use that hack elsewhere :) 

I'm fine with this NFC change but be aware that I'm not gonna approve using `#include "../../Dialect/Transforms/TargetLowering/LowerModule.h"` over `LowerToLLVM.cpp`, so we might need other PRs to look into putting those things in a proper location - note that they don't necessarily need to be public, but we could use cmake to pass in proper include dirs, etc.